proto/expr.proto

```diff
@@ -449,6 +449,13 @@ message WindowFrame {
   Bound start = 2;
   Bound end = 3;
   Exclusion exclusion = 4;
+
+  // The data type of order column in `RANGE` frame.
+  optional data.DataType order_data_type = 10;
+  // The order type of order column in `RANGE` frame.
+  optional common.OrderType order_type = 15;
+  // The data type of offset value in `RANGE` frame.
+  optional data.DataType offset_data_type = 20;
```

Comment:
It would look better if we split the the `oneof offset` as well. IIUC, the `integer` there was designed for row frames while `datum` was designed for range frames. The result will be like: 
```
oneof Frame {
    RowFrame row = 1;
    RangeFrame range = 2;
}
message RowFrame {...}
message RangeFrame {...}
```

However I also consider it's hard to keep compatibablity in this way...

Reply:
You're right! I adjusted the proto messages in the latest commit https://github.com/risingwavelabs/risingwave/pull/14648/commits/4f49c79077f2c00da829cb0820f6741298eb05ed to deprecate the old fashion and align the messages with Rust types. Seems not that hard to maintain compatibility.